lib/scientist/experiment.rb

```diff
@@ -318,4 +326,12 @@ def generate_result(name)
     control = observations.detect { |o| o.name == name }
     Scientist::Result.new(self, observations, control)
   end
+
+  private
+
+  def marshalize(block)
+    def block._dump(_)
```

Comment:
Could you add a check first before overriding? `respond_to?(:_dump) || respond_to?(:_dump_data)`

Reply:
Great idea. I'm working to add that.

Reply:
@zerowidth I added a check to this before overriding. Please let me know if you have any other feedback. 🙏